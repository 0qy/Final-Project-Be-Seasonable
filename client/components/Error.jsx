import React from 'react'
import { connect } from 'react-redux'

function ErrorComponent (props) {
  return props.error
    ? <div data-testid={'error'} style={{ margin: '100px', textAlign: 'center', color: 'red' }}>
      <h1><b>{props.error}</b></h1>
      <img style={{ width: '200px' }} src='/waiting.gif' alt='🌶️ HOLD UP BUDDY 🌶️' />
    </div>
    : null
}

function mapStateToProps (state) {
  return {
    error: state.error
  }
}

export default connect(mapStateToProps)(ErrorComponent)
