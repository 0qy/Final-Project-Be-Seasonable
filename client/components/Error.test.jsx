import React from 'react'
import '@testing-library/jest-dom/extend-expect'

import { renderWithRedux } from '../tests/test-utils'
import Error from './Error'

describe('Error component redux tests', () => {
  it('true renders error message', () => {
    const { getByTestId } = renderWithRedux(<Error />, {
      initialState: {
        error: true
      }
    })
    const error = getByTestId('error')
    expect(error).toBeVisible()
  })

  it('false renders nothing', () => {
    const { container } = renderWithRedux(<Error />, {
      initialState: {
        error: false
      }
    })
    expect(container.textContent).toBe('')
  })
})

describe('Snapshot tests - Error component', () => {
  it('renders correctly with an error', () => {
    const { container } = renderWithRedux(
      <Error />, {
        initialState: {
          error: 'Test error message'
        }
      }
    )

    expect(container).toMatchSnapshot()
  })
  it('renders correctly with no error', () => {
    const { container } = renderWithRedux(
      <Error />
    )

    expect(container).toMatchSnapshot()
  })
})
