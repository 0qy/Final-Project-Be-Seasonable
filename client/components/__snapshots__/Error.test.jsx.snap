// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests - Error component renders correctly with an error 1`] = `
<div>
  <div
    data-testid="error"
    style="margin: 100px; text-align: center;"
  >
    <h2>
      Oops. We slipped up!
    </h2>
    <p>
      Test error message
    </p>
    <img
      alt="🌶️ HOLD UP BUDDY 🌶️"
      src="/error.png"
      style="width: 200px;"
    />
  </div>
</div>
`;

exports[`Snapshot tests - Error component renders correctly with no error 1`] = `<div />`;
